src/pcre2_compile.c

```diff
@@ -6170,21 +6176,193 @@ for (;; pptr++)
 
     if ((*pptr & CLASS_IS_ECLASS) != 0)
       {
+      eclass_op_info op_info;
+      PCRE2_SIZE previous_length = (lengthptr != NULL)? *lengthptr : 0;
+      BOOL allbitsone = TRUE;
+      BOOL allbitszero = TRUE;
+
       previous = code;
       *code++ = OP_ECLASS;
       code += LINK_SIZE;
-      if (!PRIV(compile_class_nested)(options, xoptions, &pptr, &code,
-                                      errorcodeptr, cb, lengthptr))
+      *code++ = 0;  /* Flags, currently zero. */
+      if (!PRIV(compile_class_nested)(options, xoptions, FALSE, &pptr, &code,
+                                      &op_info, errorcodeptr, cb, lengthptr))
         return 0;
-      PUT(previous, 1, (int)(code - previous));
 
-      zeroreqcu = reqcu;
-      zeroreqcuflags = reqcuflags;
-      if (firstcuflags == REQ_UNSET) firstcuflags = REQ_NONE;
-      zerofirstcu = firstcu;
-      zerofirstcuflags = firstcuflags;
+      if (lengthptr != NULL)
+        {
+        *lengthptr += code - previous;
+        code = previous;
+        /* (*lengthptr - previous_length) now holds the amount of buffer that
+        we require to make the call to compile_class_nested() with
+        lengthptr = NULL, and including the (1+LINK_SIZE+1) that we write out
+        before that call. */
+        }
+
+      /* Do some useful counting of what's in the bitmap. */
+      for (int i = 0; i < 8; i++)
+        if (op_info.bits.classwords[i] != 0xffffffff)
+          { allbitsone = FALSE; break; }
```

Comment:
The syntax for these is not changed.

Reply:
I did change the indentation. Is it wrong to put it all on one line?

Reply:
I have split the code onto multiple lines.